p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,75 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const initialState = fieldsList.map(field=> ({...field, ready:false, value:""}))
+
 const Form = () => {
-  
+
+  const [fieldsReady, setFieldsReady] = useState(false)
+  const [listo, setListo] = useState('Not ready to send')
+  const [fieldsForm, setFieldsForm] = useState(initialState)
+
+  useEffect(()=> {
+    setFieldsReady(true)
+    setListo('Ready to send')
+    fieldsForm.map((field) =>{
+      if (!field.ready) {
+        setFieldsReady(false)
+        setListo('Not ready to send')
+      }
+    });
+  }, [fieldsForm]);
+
+  const handleOnChange = (e) => {
+    const { name, value } = e.target
+    console.log (`campo ${name} valor ${value}`)
+
+    setFieldsForm(newFieldForms => {
+      return newFieldForms.map(field => {
+        return field.name===name ? {...field, value: value, ready : isFieldValid(value)}:field
+      })
+    })
+
+    // setFieldsForm((current) => ({
+    //   ...current, ...{[name]:value}
+    //   }
+    // ))
+  }
+
+  const handleClickClear = () => {
+    console.log('Clear')
+    setFieldsForm(initialState)
+  }
+
+  const handleClickSubmit = () => {
+    window.alert("Send button clicked");
+  }
+   
   return (
     <Card width={"50%"}>
       <div className="form">
-        <Input label="this is label" />
+        
+        {console.log (fieldsForm)}
+        {fieldsForm.map((field) =>(
+          <Input
```

Comment:
if you see closely, you are assigning variables to the same name variables, so you can do something like this. 

```
<Input 
  key={field.name} 
  onChange={handleChange}
  {...field}
/>
```

this may depend on the eslint configuration, because you can register the rule for no spread operators in components, but most of the time there is not problem with that

Reply:
I got it